File: client/iih_stub/iih.ci_stub/iih.ci.ord/i/ICiOrdVerifyService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iih.ci.ord.dto.orderverify.d;
using xap.mw.coreitf.d;

namespace iih.ci.ord.i
{
    public interface ICiOrdVerifyService
    {
        /// <summary>
        /// 医嘱审核检索患者列表
        /// </summary>
        /// <param name="cond"></param>
        /// <returns></returns>
        OrderVerifyPatDTO[] getVerifyPat(OrderVerifyCondDTO cond);

        /// <summary>
        /// 通过医嘱id检索医嘱信息
        /// </summary>
        /// <param name="idOrs"></param>
        /// <returns></returns>
        OrderVerifyDTO[] getPatOrderByIds(String idOrs);

        /// <summary>
        /// 系统自动审核医嘱
        /// </summary>
        /// <param name="orderIds"></param>
        /// <returns></returns>
        OrderVerifyDTO[] verifyOrderBySys(string[] orderIds);

        /// <summary>
        /// 医嘱审核确认
        /// </summary>
        /// <param name="orders"></param>
        void verifyOrder(OrderVerifyDTO[] orders);

        /// <summary>
        /// 医生核对异常医嘱
        /// </summary>
        /// <param name="ordId"></param>
        /// <param name="ifAccept"></param>
        /// <param name="explain"></param>
        void confirmVerifyResult(string ordId, FBoolean ifAccept, string explain);
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/i/ICiOrdVerifyService.java
package iih.ci.ord.i;

import iih.ci.ord.dto.orderverify.d.OrderVerifyCondDTO;
import iih.ci.ord.dto.orderverify.d.OrderVerifyDTO;
import iih.ci.ord.dto.orderverify.d.OrderVerifyPatDTO;
import xap.mw.core.data.BizException;
import xap.mw.coreitf.d.FBoolean;

/**
 * 医嘱审核服务接口
 * @author ly
 *
 */
public interface ICiOrdVerifyService {
		
	/**
	 * 门诊医嘱审核检索患者医嘱
	 * @param cond 检索条件
	 * @return
	 * @throws BizException
	 */
	@Deprecated
	public abstract OrderVerifyPatDTO[] getVerifyPat(OrderVerifyCondDTO cond) throws BizException;
	
	/**
	 * 通过医嘱id检索医嘱信息
	 * @param idOrs
	 * @return
	 * @throws BizException
	 */
	public abstract OrderVerifyDTO[] getPatOrderByIds(String idOrs) throws BizException;
	
	/**
	 * 系统自动审核医嘱
	 * @param orderIds
	 * @return 系统审核信息
	 * @throws BizException
	 */
	public abstract OrderVerifyDTO[] verifyOrderBySys(String[] orderIds) throws BizException;

	/**
	 * 医嘱审核确认
	 * @param orders
	 * @throws BizException
	 */
	public abstract void verifyOrder(OrderVerifyDTO[] orders) throws BizException;
	
	/**
	 * 医生核对异常医嘱
	 * @param ordId
	 * @param ifAccept
	 * @param explain
	 * @throws BizException
	 */
	public abstract void confirmVerifyResult(String ordId, FBoolean ifAccept, String explain) throws BizException;

	/**
	 * 审核状态
	 */
	public static final String VERIFY_STATE_NOT = "0";
	public static final String VERIFY_STATE_HAS = "1";
	public static final String VERIFY_STATE_FORCE = "2";
}
